web/src/app/multiplayer/page.tsx

```diff
@@ -0,0 +1,138 @@
+"use client"
+
+import { useRouter } from 'next/navigation'
+import { useEffect, useState } from "react";
+import { Button, Table, TextInput, Title } from '@mantine/core';
+import socket from '@/scripts/SocketConnection';
+
+interface Room {
+    roomID: string;
+    players: object;
+    roomHost: string;
+}
+  
+export default function Multiplayer() {
+
+    const router = useRouter();
+
+    const [rooms, setRooms] = useState<Room[]>([]);
+    const [search, setSearch] = useState('');
+    const [filteredRooms, setFilteredRooms] = useState<Room[]>();
+
+    const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        
+        //TODO: Maybe implement a better search function?? Kinda trash/wonky
+        const { value } = event.currentTarget;
+        setSearch(value);
+
+        if (search.length != 0) {
+            const filtered = rooms.filter((room) =>
+                room.roomID.toLowerCase().includes(value.toLowerCase())
+            );
+            setFilteredRooms(filtered);
+        } 
+    };
+
+    useEffect(() => {
+        setFilteredRooms(rooms)
+    }, [rooms])
+
+    const rows = filteredRooms?.map((element) => {
+        return (
+            <Table.Tr key={element.roomID}>
+                <Table.Td className="pl-0 w-12">{element.roomID}</Table.Td>
+                <Table.Td className="pl-14" style={{ textAlign: 'left' }}>{Object.keys(element.players).length} / 4</Table.Td>
+                <Table.Td className="pl-8" > 
+                    <Button variant="outline" color="gray" radius="md" onClick={() => joinRoom(element.roomID)}>Join</Button>
+                </Table.Td>
+            </Table.Tr>
+        );
+    });
+
+    const joinRoom = (roomID : string) => {
+        router.push(roomID)
+    }
+
+    const createRoom = () => {
+        let newRoomID = String(Math.floor(Math.random() * 10000) + 10000).toString().substring(1);
+        router.push(newRoomID);
+    }
+
+    useEffect(() => {
+
+        socket.on('connect', () => {
+            socket.emit('getRooms')
+        });
+
+        socket.on('getAllRooms', (allRooms: Room[])=> {
+            setRooms(allRooms)
+        })
+
+        socket.on('updateMultiplayerRoomList', (newRooms : Room[]) => {
```

Comment:
since we get the complete list of rooms on each call to the server, we don't need to update the list like this. We can just use the getAllRooms call and replace the updateMultiplayerRoomList calls in server.ts with getAllRooms

Reply:
No we need this because of dynamic updates such as new users creating a room or a room being deleted

Reply:
I thought you meant something else, but I got it.

Reply:
when a room is created or deleted, the following line of code is called:

`socket.broadcast.emit('getAllRooms', this.#getRoomsDTO())`

this broadcasts the entire updated list, which will entirely replace the currently displayed list